subprojects/docs/src/docs/userguide/multi_project_builds.adoc

```diff
@@ -48,403 +48,660 @@ Configuration on demand mode attempts to configure only projects that are releva
+----
 
 As we have said, we usually prefer to put project specific behavior into the build script of this project. Let's refactor and also add some project specific behavior to the `krill` project.
 
-++++
-<sample id="multiprojectSpreadSpecifics" dir="userguide/multiproject/spreadSpecifics/water" includeLocation="true" title="Defining specific behaviour for project krill">
-                <layout>
-                    build.gradle
-                    settings.gradle
-                    bluewhale/build.gradle
-                    krill/build.gradle
-                </layout>
-                <sourcefile file="settings.gradle"/>
-                <sourcefile file="bluewhale/build.gradle"/>
-                <sourcefile file="krill/build.gradle"/>
-                <sourcefile file="build.gradle"/>
-                <output args="-q hello"/>
-            </sample>
-++++
+=== Example: Defining specific behaviour for project krill
+
+.Project layout
+----
+.
+├── bluewhale
+│   └── build.gradle
+├── build.gradle
+├── krill
+│   └── build.gradle
+└── settings.gradle
+----
+
+[source.multi-language-sample,groovy]
+.settings.gradle
+----
+include::{samplesPath}/userguide/multiproject/spreadSpecifics/water/settings.gradle[]
+----
+
+[source.multi-language-sample,groovy]
+.bluewhale/build.gradle
+----
+include::{samplesPath}/userguide/multiproject/spreadSpecifics/water/bluewhale/build.gradle[]
+----
+
+[source.multi-language-sample,groovy]
+.krill/build.gradle
+----
+include::{samplesPath}/userguide/multiproject/spreadSpecifics/water/krill/build.gradle[]
+----
+
+[source.multi-language-sample,groovy]
+.build.gradle
+----
+include::{samplesPath}/userguide/multiproject/spreadSpecifics/water/build.gradle[]
+----
+
+.Output of `gradle -q hello`
+----
+> gradle -q hello
+include::{samplesPath}/userguide/multiproject/spreadSpecifics/water/multiprojectSpreadSpecifics.out[]
+----
 
 
 [[sub:project_filtering]]
-==== Project filtering
+=== Project filtering
 
 To show more of the power of configuration injection, let's add another project called `tropicalFish` and add more behavior to the build via the build script of the `water` project.
 
 
 [[ssub:filtering_by_name]]
-===== Filtering by name
-
-
-++++
-<sample id="multiprojectAddTropical" dir="userguide/multiproject/addTropical/water" includeLocation="true" title="Adding custom behaviour to some projects (filtered by project name)">
-                    <layout>
-                        build.gradle
-                        settings.gradle
-                        bluewhale/build.gradle
-                        krill/build.gradle
-                        tropicalFish/
-                    </layout>
-                    <sourcefile file="settings.gradle"/>
-                    <sourcefile file="build.gradle"/>
-                    <output args="-q hello"/>
-                </sample>
-++++
+==== Filtering by name
+
+=== Example: Adding custom behaviour to some projects (filtered by project name)
```

Comment:
:x: Missing note admonition about where to find the sample in the `-all` distribution, same as above